lib/patch-series.ts

```diff
@@ -328,23 +328,19 @@ export class PatchSeries {
 
     protected static encodeSender(sender: string): string {
         const encoded = encodeWords(sender);
-        const dot = encoded.indexOf(".");
-        if (dot < 0) {
+
+        /* Don't quote if already quoted */
+        if (encoded.startsWith("\"") && encoded.indexOf("\" <") > 0) {
             return encoded;
         }
 
-        let endOfName = encoded.indexOf("<");
-        if (endOfName < 0) {
-            endOfName = encoded.length;
-        } else {
-            while (endOfName > 0 && encoded[endOfName - 1] === " ") {
-                endOfName--;
-            }
-        }
-        if (dot >= endOfName) {
+        const match =
+            encoded.match(/^([^<]*[()<>[\]:;@\\,."][^<]*?)(\s*)(<.*)/);
```

Comment:
This is a bit difficult to reason about but I _think_ it's correct. To make it more obvious that this works as intended, perhaps in addition to the test @derrickstolee suggested, test cases could be added which verify that these do not get quoted:

```
name <email>
my name <email>
excited! <email> 
```

and these get quoted:

```
mr. name <email>
foo [bar] name <email>
bob@obo <email>
blug <nobody <email>
```

(I'm not sure the last one has any value; it almost certainly isn't even legal so perhaps it doesn't deserve to be tested.)

Reply:
I added all of them, and also one containing a backslash.

Reply:
I added all of them, and also one containing a backslash.

Reply:
I added all of them, and also one containing a backslash.